### PR TITLE
Clean-up of 45_carbonprice module

### PR DESCRIPTION
**Pull request by laurinks**
## Purpose of this PR

Clean-up of 45_carbonprice module: 
- Remove realizations `diffCurvPhaseIn2Lin` (superseded by `diffLin2Lin`), `exponential` (superseded by `diffExp2Lin`) and `linear` (superseded by `diffLin2Lin`),
- Consistent update of scenario config files (affects project files),
- Improved documentation of `diffLin2Lin` and `diffExp2Lin` in `main.gms`.

## Type of change

(Make sure to delete from the Type-of-change list the items not relevant to your PR)

- [x] Refactoring
- [x] Minor change (default scenarios show no differences)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I explained my changes within the PR, particularly in hard-to-understand areas
- [x] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [ ] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [x] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [x] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

### CODE REVIEW

File: config/scenario_config_21_EU11_ECEMF.csv
Comment (laurinks) [inline config/scenario_config_21_EU11_ECEMF.csv:1]:
```
File comparison: /p/projects/rd3mod/github/repos/remindmodel/remind/develop/config/scenario_config_21_EU11_ECEMF.csv -> config/scenario_config_21_EU11_ECEMF.csv
Columns deleted: -
Columns added:   -
Renamed columns: cm_co2_tax_2020 -> cm_co2_tax_startyear
Renamed rows:    -

Changes in the scenarios:
~ xx_DIAG-NZero:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_Nzero:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_NZero-LimBio:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_NZero-LimCCS:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_NZero-LimNuclear:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-CP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-CP-LimBio:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-CP-LimCCS:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-CP-LimNuclear:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-REG:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-REG-LimBio:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-REG-LimCCS:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_OPT-REG-LimNuclear:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-CP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-CP-LimBio:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-CP-LimCCS:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-CP-LimNuclear:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-REG:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-REG-LimBio:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-REG-LimCCS:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP5_RAP-REG-LimNuclear:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ test_WP1_Nzero_DEU:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_Nzero_highPrecision_10tCO2eq:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_Nzero_highPrecision_5tCO2eq:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_Nzero_highPrecision_1tCO2eq:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ xx_WP1_Nzero_highPrecision_0p4tCO2eq:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)

$allwarnings
NULL
```

File: config/scenario_config_DeepEl.csv
Comment (laurinks) [inline config/scenario_config_DeepEl.csv:1]:
```
File comparison: /p/projects/rd3mod/github/repos/remindmodel/remind/develop/config/scenario_config_DeepEl.csv -> config/scenario_config_DeepEl.csv
Columns deleted: -
Columns added:   -
Renamed columns: cm_co2_tax_2020 -> cm_co2_tax_startyear
Renamed rows:    -

Changes in the scenarios:
~ SSP2-PkBudg500-DeepEl:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2-PkBudg1150-DeepEl:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)

$allwarnings
NULL
```

File: config/scenario_config_EL2p0DeepDive.csv
Comment (laurinks) [inline config/scenario_config_EL2p0DeepDive.csv:1]:
```
File comparison: /p/projects/rd3mod/github/repos/remindmodel/remind/develop/config/scenario_config_EL2p0DeepDive.csv -> config/scenario_config_EL2p0DeepDive.csv
Columns deleted: -
Columns added:   -
Renamed columns: cm_co2_tax_2020 -> cm_co2_tax_startyear
Renamed rows:    -

Changes in the scenarios:
~ SSP2EU-noDSPkB500-noDS_betax_DeepDive:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB500-DS_betax_DeepDive:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB650-noDS_betax_DeepDive:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB650-DS_betax_DeepDive:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB1050-noDS_betax_DeepDive:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB1050-DS_betax_DeepDive:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB500-noDS_betax_AgMIP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB500-DS_betax_AgMIP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB650-noDS_betax_AgMIP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB650-DS_betax_AgMIP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB1050-noDS_betax_AgMIP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB1050-DS_betax_AgMIP:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB500-noDS_betax_DeepDive_noNDC:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB500-DS_betax_DeepDive_noNDC:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB650-noDS_betax_DeepDive_noNDC:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB650-DS_betax_DeepDive_noNDC:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-noDSPkB1050-noDS_betax_DeepDive_noNDC:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ SSP2EU-DSPkB1050-DS_betax_DeepDive_noNDC:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)

$allwarnings
NULL
```

File: config/scenario_config_ELEVATE4p4.csv
Comment (laurinks) [inline config/scenario_config_ELEVATE4p4.csv:1]: File comparison: /p/projects/rd3mod/github/repos/remindmodel/remind/develop/config/scenario_config_ELEVATE4p4.csv -> config/scenario_config_ELEVATE4p4.csv
Columns deleted: -
Columns added:   -
Renamed columns: cm_co2_tax_2020 -> cm_co2_tax_startyear
Renamed rows:    -

Changes in the scenarios:

$allwarnings
NULL

File: config/scenario_config_NGFS_v5.csv
Comment (laurinks) [inline config/scenario_config_NGFS_v5.csv:1]:
```
File comparison: /p/projects/rd3mod/github/repos/remindmodel/remind/develop/config/scenario_config_NGFS_v5.csv -> config/scenario_config_NGFS_v5.csv
Columns deleted: -
Columns added:   -
Renamed columns: cm_co2_tax_2020 -> cm_co2_tax_startyear
Renamed rows:    -

Changes in the scenarios:
~ o_1p5c:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_lowdem:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_2c:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_delfrag:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_rap:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_KLW_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_lowdem_KLW_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_2c_KLW_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_delfrag_KLW_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_rap_KLW_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_d95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_d50high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_d95high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_d50_cpricereg:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_d95_cpricereg:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_dni50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_1p5c_dni95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_lowdem_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_lowdem_d95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_lowdem_d50high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_lowdem_d95high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ o_2c_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_d95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_d50high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_d95high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_d50_cpricereg:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_d95_cpricereg:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_dni50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ o_2c_dni95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_delfrag_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_delfrag_d95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_delfrag_d50high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_delfrag_d95high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
~ d_rap_d50:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ d_rap_d95:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ d_rap_d50high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)
~ d_rap_d95high:
    carbonprice: diffCurvPhaseIn2Lin -> diffLin2Lin (default: none)
    cm_co2_tax_2020 -> cm_co2_tax_startyear: 200 -> 150 (default: -1)

$allwarnings
NULL
```

File: main.gms [hunk 1]
```diff
@@ -371,17 +371,27 @@ $setglobal emicapregi  none           !! def = none
 *'
 *' This module defines the carbon price pm_taxCO2eq, with behaviour across regions governed by similar principles (e.g. global targets, or all following NDC or NPi policies).
 *'
-*' * (none): no tax policy (combined with all emiscens except emiscen = 9)
-*' * (exponential): [please use new diffExp2Lin with cm_co2_tax_spread = 1 and iterative_target_adj = 5 for exponential carbon pricing until end of century (without regional differentiation)] 4.5% exponential increase over time of the tax level in 2020 set via cm_co2_tax_2020 (combined with emiscen = 9 and cm_co2_tax_2020>0)
-*' * (expoLinear): 4.5% exponential increase until c_expoLinear_yearStart, transitioning into linear increase thereafter
+*' * (diffExp2Lin) and (diffLin2Lin): standard carbonprice realizations for ambitious climate policy scenarios [REMIND default for peak budget runs: diffLin2Lin in combination with iterative_target_adj = 9],
```
Comment (laurinks) [inline main.gms:374]: Improved documentation of (diffExp2Lin) and (diffLin2Lin).

File: main.gms [hunk 2]
```diff
@@ -371,17 +371,27 @@ $setglobal emicapregi  none           !! def = none
...
 *' * (NDC): implements a carbon price trajectory consistent with the NDC targets (up to 2030) and a trajectory of comparable ambition post 2030 (1.25%/yr price increase and regional convergence of carbon price). Choose version using cm_NDC_version "2023_cond", "2023_uncond", or replace 2023 by 2022, 2021 or 2018 to get all NDC published until end of these years.
 *' * (NPi): National Policies Implemented, extrapolation of historical (until 2020) carbon prices
+*' * (none): no tax policy (combined with all emiscens except emiscen = 9)
+
+***  (exponential) is superseded by (diffExp2Lin): For a globally uniform, exponentially increasing carbonprice path until end of century [in combination with cm_iterative_target_adj = 0 or 5], set cm_co2_tax_spread = 1, set cm_peakBudgYr = 2110, and choose the initial carbonprice in cm_startyear via cm_co2_tax_startyear.
```
Comment (laurinks) [inline main.gms:392]: Additional explanations on how to obtain behavior of superseded realizations (exponential) and (linear)

File: main.gms [hunk 3]
```diff
@@ -498,17 +507,10 @@ parameter
 *' *  (1): BAU
 *' *  (4): emission time path
 *' *  (6): budget
-*' *  (9): tax scenario (requires running module 21_tax "on"), tax level controlled by module 45_carbonprice and cm_co2_tax_2020, other GHG etc. controlled by cm_rcp_scen
+*' *  (9): tax scenario (requires running module 21_tax "on"), tax level controlled by module 45_carbonprice and cm_co2_tax_startyear, other GHG etc. controlled by cm_rcp_scen
 *' *  (10): used for cost-benefit analysis
 *' *JeS* WARNING: data for cm_emiscen 4 only exists for multigas_scen 2 bau scenarios and for multigas_scen 1
 *'
-parameter
```
Comment (laurinks) [inline main.gms:505]: Old switch, no longer used.

File: modules/45_carbonprice/diffLin2Lin/datainput.gms [hunk 4]
```diff
@@ -14,7 +14,7 @@
 ***                                         carbon price of developed countries increases linearly with fixed annual increase given by cm_taxCO2inc_after_peakBudgYr (default = 0, i.e. constant)
 ***       (with iterative_target_adj = 5):  carbon price of developed countries keeps increasing linearly (with same slope) until end of century, i.e. no change after peak year 
 ***       (with iterative_target_adj = 0):  after year given by cm_peakBudgYr (default = 2050), carbon price of developed countries increases linearly with fixed annual increase given by cm_taxCO2inc_after_peakBudgYr (default = 0, i.e. constant),
-***                                         for linearly increasing carbon price (with same slope) until end of century, set cm_peakBudgYr = 2100         
+***                                         for linearly increasing carbon price (with same slope) until end of century, set cm_peakBudgYr = 2110
```
Comment (laurinks) [inline modules/45_carbonprice/diffLin2Lin/datainput.gms:17]: Consistent implementation in (diffLin2Lin) and (diffExp2Lin): If linear/exponential increase is selected until end of century (e.g. with cm_iterative_target_adj = 5), carbonprices are set constant from 2110 onward.

File: modules/81_codePerformance/on/realization.gms [hunk 5]
```diff
@@ -8,7 +8,6 @@
 
 
 *' @description BAU, tax30, and tax150 runs are set in a loop of 30 runs in total.
-*' The realization needs the realization "exogenous" of the 45_carbonprice module
```
Comment (laurinks) [inline modules/81_codePerformance/on/realization.gms:11]: Outdated documentation

File: modules/81_codePerformance/on/presolve.gms
Comment (laurinks) [inline modules/81_codePerformance/on/presolve.gms:1]: Not sure if this module is still in use. Nonetheless cleaned it up. Previously, it just used a copy-paste version of the 45_carbonprice/exponential with exponentially increasing carbonprices from 2010 onward (with the 2020 value set to 0,30,150). For simplicity, I reduced it to one line of code. Now, it is still exponentially increasing from 2010 onward but the startyear (default = 2005) value is set to 0,30,150.

File: main.gms (hunk 2 above)
Comment (orichters) [reply main.gms:392]: That is helpful, I think.

**Review by orichters — APPROVED** [review]
I have to admit I did not check everything in detail, but overall it looks very convincing. Thanks for investing in checking documentation and tutorials, that is very appreciated!

**Review by LaviniaBaumstark — APPROVED** [review]
thanks for all that cleaning!
the scenario:config.csv file is already using teh new formulation, right?

**Comment by laurinks** [comment]
Thanks for the reviews! @LaviniaBaumstark : Yes, diffLin2Lin was introduced as the REMIND default in [PR#1747](https://github.com/remindmodel/remind/pull/1747). That's why the clean-up changes now only affect some project configs.

**Pull request merged**